lib/fetch/index.js

```diff
@@ -1808,10 +1808,10 @@ async function httpNetworkFetch (
         fetchParams.controller.controller = controller
       },
       async pull (controller) {
-        await pullAlgorithm(controller)
+        pullAlgorithm(controller)
```

Comment:
!!! This is wrong? needs a return at least?

Reply:
It was before not returning anything. So it is fire and forget? So why await anyway?

Reply:
wasn't it at least informing the caller that the operation is done? I'm not sure where this is used, this can't possible cause racing conditions or break someone's tests?